backend/regalloc/regalloc_irc_state.ml

```diff
@@ -150,22 +148,22 @@ let[@inline] add_initial_one state reg =
 
 let[@inline] add_initial_list state regs =
   List.iter regs ~f:(fun reg ->
-      Reg.Tbl.replace state.reg_work_list reg WorkList.Initial;
+      Reg.Tbl.replace state.reg_work_list reg RegWorkList.Initial;
       Reg.Tbl.replace state.reg_color reg None;
       Reg.Tbl.replace state.reg_alias reg None;
       Reg.Tbl.replace state.reg_interf reg [];
       Reg.Tbl.replace state.reg_degree reg 0;
       Doubly_linked_list.add_begin state.initial reg)
 
 let[@inline] reset state ~new_inst_temporaries ~new_block_temporaries =
-  let unknown_reg_work_list (rwl : RegWorkList.t) : unit =
-    RegWorkList.iter rwl ~f:(fun reg ->
-        Reg.Tbl.replace state.reg_work_list reg WorkList.Unknown_list)
+  let unknown_reg_work_list (rwl : RegWorkListSet.t) : unit =
+    RegWorkListSet.iter rwl ~f:(fun reg ->
+        Reg.Tbl.replace state.reg_work_list reg RegWorkList.Unknown_list)
   in
   let unknown_instruction_work_list (iwl : InstructionWorkList.t) : unit =
```

Comment:
I think this can be deleted since the table will be cleared. Should we also do the same for `reg_work_list`?